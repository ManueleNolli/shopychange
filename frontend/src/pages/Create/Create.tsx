import React from 'react'

import { Box, Center, Flex, Heading } from '@chakra-ui/react'
import ShadowButton from '../../components/ShadowButton/ShadowButton'
import { colors } from '../../styles/Colors'
import { useNavigate } from 'react-router-dom'

export default function Create() {
  const navigate = useNavigate()

  const onNavigation = (path: string) => {
    navigate(`/create/${path}`)
  }

  return (
    <Box>
      <Center>
        <Heading>Create NFTs</Heading>
      </Center>
      <Center mt="5vh">
        <Flex direction="column" justifyContent={'center'}>
          <ShadowButton
            label={'Create Single NFT'}
            bgColor={`${colors.blueChakraColor}.400`}
            textColor={'white'}
            shadowColor={colors.blueShadowColor}
            hoverColor={`${colors.blueChakraColor}.500`}
            focusColor={`${colors.blueChakraColor}.500`}
            buttonProps={{
              width: { base: '60vw', md: '40vw', '2xl': '20vw' },
              height: '10vh',
            }}
            onClick={() => onNavigation('nft')}
          />
          <ShadowButton
            label={'Create Collection'}
            bgColor={`${colors.blueChakraColor}.400`}
            textColor={'white'}
            shadowColor={colors.blueShadowColor}
            hoverColor={`${colors.blueChakraColor}.500`}
            focusColor={`${colors.blueChakraColor}.500`}
            buttonProps={{
              mt: '5vh',
              width: { base: '60vw', md: '40vw', '2xl': '20vw' },
              height: '10vh',
            }}
            onClick={() => onNavigation('collection')}
          />
        </Flex>
      </Center>
    </Box>
  )
}
