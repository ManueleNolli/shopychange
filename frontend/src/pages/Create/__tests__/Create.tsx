import React from 'react'
import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Create from '../Create'

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}))

describe('Create', () => {
  test('renders correctly', () => {
    const tree = render(<Create />, { wrapper: BrowserRouter })
    expect(tree.container).toMatchSnapshot()
  })

  test('should navigate to /create/nft when Create Single NFT button is clicked', () => {
    render(<Create />, { wrapper: BrowserRouter })
    const createSingleNFTButton = screen.getByText('Create Single NFT')
    expect(createSingleNFTButton).toBeInTheDocument()
    createSingleNFTButton.click()
    expect(mockNavigate).toHaveBeenCalledWith('/create/nft')
  })

  test('should navigate to /create/collection when Create Collection button is clicked', () => {
    render(<Create />, { wrapper: BrowserRouter })
    const createCollectionButton = screen.getByText('Create Collection')
    expect(createCollectionButton).toBeInTheDocument()
    createCollectionButton.click()
    expect(mockNavigate).toHaveBeenCalledWith('/create/collection')
  })
})
