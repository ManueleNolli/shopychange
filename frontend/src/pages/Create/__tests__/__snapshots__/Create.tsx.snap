// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <h2
        class="chakra-heading css-0"
      >
        Create NFTs
      </h2>
    </div>
    <div
      class="css-1ffdipv"
    >
      <div
        class="css-zg1vud"
      >
        <button
          class="chakra-button css-rbo9mh"
          data-testid="onClick-button"
          style="white-space: initial;"
          type="button"
        >
          Create Single NFT
        </button>
        <button
          class="chakra-button css-csrfg1"
          data-testid="onClick-button"
          style="white-space: initial;"
          type="button"
        >
          Create Collection
        </button>
      </div>
    </div>
  </div>
</div>
`;
